sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -442,7 +442,11 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   override def uncacheTable(tableName: String): Unit = {
     val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
     val cascade = !sessionCatalog.isTemporaryTable(tableIdent)
-    sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(tableName), cascade)
+    val wasCached = Try(sparkSession.catalog.isCached(tableIdent.unquotedString)).getOrElse(false)
```

Comment:
Why use Try? Will `isCached` throw exception?

Reply:
Btw, `isCached` is also in `CatalogImpl`. Why to access it via `sparkSession.catalog`?

Reply:
@viirya Thank you! 
I didn't notice  the `isCached `method in `CatalogImpl`. Therefore, I mimicked the usage in `tables.scala` to avoid exceptions.
  In` tables.scala`, 
```
      // If an exception is thrown here we can just assume the table is uncached;
      // this can happen with Hive tables when the underlying catalog is in-memory.
      val wasCached = Try(sparkSession.catalog.isCached(oldName.unquotedString)).getOrElse(false)
```
Now I am modifying to use the `isCached  `provided by `CatalogImpl ` without considering this exception.